### PR TITLE
Changed all the classes into interfaces.

### PR DESCRIPTION
**Pull request by jojotastic777**
I think this is appropriate, seeing as none of the "classes" could be used with `new`

### CODE REVIEW

File: typefall.d.ts [hunk 1]
```diff
@@ -96,48 +96,48 @@ declare interface IVector {
...
+declare interface IAngle {
+    p: number;
+    y: number;
+    r: number;
+    pitch: number;
+    yaw: number;
+    roll: number;
 
-    public constructor(pitch: number, yaw: number, roll: number);
+    constructor(pitch: number, yaw: number, roll: number);
```
Comment (Ehntae) [inline typefall.d.ts]: Why is there a constructor here?

File: typefall.d.ts [hunk 2]
```diff
@@ -181,139 +181,139 @@ declare interface IColor {
...
-    public setSolid(solid: boolean): void;
-    public setSubMaterial(index: number, material: string): void;
-    public setTrails(startSize: number, endSize: number, length: number, material: string, color: IColor, attachmentID?: number, additive?: boolean): void;
-    public setVelocity(velocity: IVector): void;
-    public translateBoneToPhysBone(id: number): number;
-    public translatePhysBoneToBone(id: number): number;
-    public unparent(): void; // ? Does this not take any argments? It doesn't according to the docs, anyways.
-    public worldToLocal(IVector: IVector): IVector;
-    public worldToLocalIAngles(IAngle: IAngle): IAngle;
+declare interface IEntity {
```
Comment (Ehntae) [inline typefall.d.ts:184]: This is good.

File: typefall.d.ts (hunk 1 above)
Comment (jojotastic777) [reply typefall.d.ts]: oh I totally missed that. also I don't think that one was my fault. Didn't you do the Angle one?

EDIT: Yep. just checked. That's totally your fault.

Comment (jojotastic777) [reply typefall.d.ts]: I'll remove it.

Comment (Ehntae) [reply typefall.d.ts]: The refactor script wasn't setup to remove constructors from classes. Remove this line in your commit and I'll pull changes.

Comment (jojotastic777) [reply typefall.d.ts]: Alright, done.

**Pull request merged**